test/WakuRlnV2.t.sol

```diff
@@ -43,6 +46,103 @@ contract WakuRlnV2Test is Test {
             w.root(),
             13_801_897_483_540_040_307_162_267_952_866_411_686_127_372_014_953_358_983_481_592_640_000_001_877_295
         );
+        (uint32 fetchedUserMessageLimit2, uint32 index2, uint256 rateCommitment2) =
+            w.idCommitmentToMetadata(idCommitment);
+        assertEq(fetchedUserMessageLimit2, userMessageLimit);
+        assertEq(index2, 0);
+        assertEq(rateCommitment2, rateCommitment);
+        vm.resumeGasMetering();
+    }
```

Comment:
we should probably add a kats here for the merkleProof elements. left it out since we'd be basically testing the upstream library, but I can add it in a follow up PR